### PR TITLE
feat: test for milestone3

### PR DESCRIPTION
**Pull request by crystaljesica**
test cases for milestone3

- test the base 64 service
- should turn utf8 to b64 and back
- should b63service works when with one word of translation
- should b63service works with a medium length translation

### CODE REVIEW

File: src/app/demo/demo.component.spec.ts [hunk 1]
```diff
@@ -239,3 +238,104 @@ function randomRange(min: number, max: number) {
...
+    劆֒񦘰ˑ{坋𹔸lǼc&񓱬񊄸Ӽ:󌈅=̹ɽ渭觙􈯶൰ȣ¡圤𹷟򱄢揋ﺝ􊃻^
+    E̶򱩀򑪟eٌϐ򔜮霗燨综*􍪻񭚴oꕃ𷴨2ҽT񺆥uR혙񗊭:򉪼񙏺ۤƓ騡
+    񱺡􌡩PȌ񸛍񩍢􅓴冖㌃ۄ𦄜7Ž⇆*zѱ澁nަvۭË́듍JҢ䆹M󩗟繶`;
+    expect(service.b64_to_utf8(service.utf8_to_b64(testUTF8))).toEqual(
+      testUTF8
+    );
+  });
+
+  // the situation when there is only one word'test'
+  it("should b63service works when with one word of translation", () => {
```
Comment (Iostream3100) [inline src/app/demo/demo.component.spec.ts:271]: maybe b64?

File: src/app/demo/demo.component.spec.ts [hunk 2]
```diff
@@ -239,3 +238,104 @@ function randomRange(min: number, max: number) {
...
+    IGlkPSJ0MGIwZDAiPgogICAgICAgICAgICAgICAgPHAgaWQ9InQwYjBkMHAwIj4KICAgICAgICAgICAgICAg
+    ICAgICA8cyBpZD0idDBiMGQwcDBzMCI+PHcgaWQ9InQwYjBkMHAwczB3MCIgQVJQQUJFVD0iREggSUggUyI+
+    dGhpczwvdz4gPHcgaWQ9InQwYjBkMHAwczB3MSIgQVJQQUJFVD0iSUggWiI+aXM8L3c+IDx3IGlkPSJ0MGIw
+    ZDBwMHMwdzIiIEFSUEFCRVQ9IkFIIj5hPC93PiA8dyBpZD0idDBiMGQwcDBzMHczIiBBUlBBQkVUPSJUIEVI
+    IFMgVCI+dGVzdDwvdz4gPHcgaWQ9InQwYjBkMHAwczB3NCIgQVJQQUJFVD0iRiBBTyBSIj5mb3I8L3c+IDx3
+    IGlkPSJ0MGIwZDBwMHMwdzUiIEFSUEFCRVQ9Ik0gQVkgTCBTIFQgT1cgTiI+bWlsZXN0b25lPC93PiA8dyBp
+    ZD0idDBiMGQwcDBzMHc2IiBBUlBBQkVUPSJHIFVIIEQiPmdvb2Q8L3c+Ljwvcz48cCBjbGFzcz0idHJhbnNs
+    YXRpb24iPui/meaYr+S4gOS4qua1i+ivlTwvcD4KICAgICAgICAgICAgICAgIDwvcD4KICAgICAgICAgICAg
+    PC9kaXY+CiAgICAgICAgPC9ib2R5PgogICAgPC90ZXh0Pgo8L1RFST4=`;
+    let translation = service.b64_to_utf8(testxml);
```
Comment (Iostream3100) [inline src/app/demo/demo.component.spec.ts:296]: unused variable

File: src/app/demo/demo.component.spec.ts (hunk 1 above)
Comment (crystaljesica) [reply src/app/demo/demo.component.spec.ts]: yes, this is a typo here. fixed

File: src/app/demo/demo.component.spec.ts (hunk 2 above)
Comment (crystaljesica) [reply src/app/demo/demo.component.spec.ts]: Thanks. delete the duplicate ones. Fixed

File: src/app/demo/demo.component.spec.ts [hunk 3]
```diff
@@ -239,3 +238,104 @@ function randomRange(min: number, max: number) {
...
+    SJDSCBISCBPVyBOIEcgQUEiPmNob25nYTwvdz4gPHcgaWQ9InQwYjBkMHAwczB3OSIgZWZmZWN0aXZlLWcycC1sYW5nPSJ1bmQiIEFSUEFCRVQ9IkNIIEhIIE9XI
+    E4gRyBBQSI+Y2hvbmdhPC93PiA8dyBpZD0idDBiMGQwcDBzMHcxMCIgZWZmZWN0aXZlLWcycC1sYW5nPSJ1bmQiIEFSUEFCRVQ9IkNIIEhIIE9XIE4gRyBBQSI+Y
+    2hvbmdhPC93PiA8dyBpZD0idDBiMGQwcDBzMHcxMSIgZWZmZWN0aXZlLWcycC1sYW5nPSJ1bmQiIEFSUEFCRVQ9IkNIIEhIIE9XIE4gRyBBQSI+Y2hvbmdhPC93P
+    jwvcz48cCBjbGFzcz0idHJhbnNsYXRpb24iPmNob25nYSBjaG9uZ2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ
+    2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ2EgY2hvbmdhIGNob25nYWNob25nYSBjaG9uZ2EgY2hvbmdhIGNob
+    25nYTwvcD4KICAgICAgICAgICAgICAgIDwvcD4KICAgICAgICAgICAgPC9kaXY+CiAgICAgICAgPC9ib2R5PgogICAgPC90ZXh0Pgo8L1RFST4=`;
+    let translation = service.b64_to_utf8(testxml);
+    expect(service.b64_to_utf8(service.utf8_to_b64(testxml))).toEqual(testxml);
+  });
+});
```
Comment (guanxiaotongyz) [inline src/app/demo/demo.component.spec.ts:337]: After fixed, all items are great. Just one remainder, the first test is similar with the other component  test, But that fine. well done.

**Review by guanxiaotongyz — APPROVED** [review]
great!

File: src/app/demo/demo.component.spec.ts (hunk 3 above)
Comment (yoxchen95) [reply src/app/demo/demo.component.spec.ts:337]: It is all done for milestone3. It is great.

**Review by yoxchen95 — APPROVED** [review]
It works fine after fixing the mistakes. Great job!

**Pull request merged**